src/app/collective-rewards/leaderboard/BuilderLeaderBoard.tsx

```diff
@@ -1,12 +1,20 @@
+import { BuildersLeaderBoardContent } from '@/app/collective-rewards/leaderboard'
 import { CycleContextProvider } from '@/app/collective-rewards/metrics'
+import { useReadBackersManager } from '@/app/collective-rewards/shared'
 import { Button } from '@/components/Button'
 import { Collapsible, CollapsibleContent, CollapsibleTrigger } from '@/components/Collapsible'
+import { Popover } from '@/components/Popover'
 import { HeaderTitle } from '@/components/Typography'
-import { BuildersLeaderBoardContent } from '@/app/collective-rewards/leaderboard'
+import { DateTime } from 'luxon'
 import { useRouter } from 'next/navigation'
 
 export const BuildersLeaderBoard = () => {
   const router = useRouter()
+  const { data: isInDistributionPeriod } = useReadBackersManager('onDistributionPeriod')
+  const { data: distributionEndTimestamp } = useReadBackersManager('endDistributionWindow', [
```

Comment:
Here we already have `useGetEndDistributionWindow`

Reply:
if we agree on using the abi-based hook I;d rather stick to that.
otherwise I'll remove it and use method-based hooks